perses/annihilation/ncmc_switching.py

```diff
@@ -135,6 +143,30 @@ def _getAvailableParameters(self, system, prefix='lambda'):
                         parameters.append(parameter_name)
         return parameters
 
+    def _updateAlchemicalState(self, context, functions, value):
+        """
+        Update alchemical state using the specified lambda value.
+
+        Parameters
+        ----------
+        context : simtk.openmm.Context
+            The Context
+        functions : dict
+            A dictionary of functions
+        value : float
+            The alchemical lambda value
+
+        TODO: Improve function evaluation to better match Lepton and be more flexible in exact replacement of 'lambda' tokens
+        Perhaps this could be done with a special integrator?
+
+        """
+        from perses.annihilation import NumericStringParser
```

Comment:
We already do, but we need some way to measure the work (potential energy change) for going from fully interacting system -> alchemically-modified system with `lambda = 1` -> alchemically-modified system with `lambda` at its starting value. We can do this by adding some more variables inside of the integrator that we can then access after the fact, though. I'll see if I can add this and remove this extra code.